lib/vsc/mympirun/pmi/mpi.py

```diff
@@ -0,0 +1,320 @@
+#
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of vsc-mympirun,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-mympirun
+#
+# vsc-mympirun is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# vsc-mympirun is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with vsc-mympirun.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Base PMI MPI class, all actual classes should inherit from this one
+
+The role of the MPI class is very limited, mainly to provide supported PMI flavour/version
+"""
+from __future__ import print_function
+
+import os
+
+from vsc.utils.fancylogger import getLogger
+from vsc.mympirun.common import MpiBase, eb_root_version, version_in_range
+from vsc.mympirun.pmi.pmi import PMIv2, PMIxv3
+
+# TODO: should be made generic somehow
+#   pmix3 pmi2 compat libraries might not work with all mpi apps
+#   but one can package the slurm pmi2 libs on different location
+#   ideally, one tests for pmix or not, and somehow tests that the pmi2 libs are not from pmix package
+AUTOMATIC_LIB = 'AUTOMATIC'
+SLURM_PMI1_LIB = 'SLURM1'
+SLURM_PMI2_LIB = 'SLURM2'
+SYSTEM_PMI1_LIB = 'SYSTEM1'
+SYSTEM_PMI2_LIB = 'SYSTEM2'
+
+PMI2LIBS = {
+    AUTOMATIC_LIB: '/automatic/via/does/not/exist',  # non-exisiting file picks up some defaults and "stuff works"
+    SLURM_PMI1_LIB: '/usr/lib64/slurmpmi/libpmi.so',
+    SLURM_PMI2_LIB: '/usr/lib64/slurmpmi/libpmi2.so',
+    SYSTEM_PMI1_LIB: '/usr/lib64/libpmi.so',  # possibly/likely pmix
+    SYSTEM_PMI2_LIB: '/usr/lib64/libpmi2.so',  # possibly/likely pmix
+}
+
+
+class MPI(MpiBase):
```

Comment:
Should this still be `MPI` rather than `PMI`?

Reply:
it is still about the mpi part (pmi replaces mpirun, not mpi)